Cython/Compiler/ExprNodes.py

```diff
@@ -208,10 +211,14 @@ def make_dedup_key(outer_type, item_nodes):
               type(node.constant_result) if node.type is py_object_type else None) if node.has_constant_result()
         else None  # something we cannot handle => short-circuit below
         for node in item_nodes
-    ]
+    ]  # Use list comprehensions intentionally
```

Comment:
Words like "intentionally" aren't helpful as comments because they lack explanation. The important part that a comment should convey is: why? *What* was the intention? Under which circumstances is it ok for Future Dev to consider this comment obsolete and change the code? When can I tear down Chesterton's Fence?
```suggestion
    ]
```